modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -33,124 +37,149 @@ final case class SnapshotRoutes[F[_]: Async, S <: Snapshot: Encoder, C: Encoder]
   prefixPath: InternalUrlPrefix,
   nodeStorage: NodeStorage[F],
   hasherSelector: HasherSelector[F],
-  snapshotTimeoutsConfig: SnapshotTimeoutsConfig
+  snapshotTimeoutsConfig: SnapshotTimeoutsConfig,
+  latestSnapshotCache: LatestSnapshotCache[F, S, C]
 ) extends Http4sDsl[F]
     with PublicRoutes[F]
     with P2PRoutes[F] {
 
   object FullSnapshotQueryParam extends FlagQueryParamMatcher("full")
 
-  // first on the list is the default - used when `Accept: */*` is requested
-  implicit def jsonEncoders[A <: AnyRef: Encoder]: List[EntityEncoder[F, A]] = List(CirceEntityEncoder.circeEntityEncoder[F, A])
-
-  private val serviceUnavailableNodeNotReady: F[Response[F]] =
-    ServiceUnavailable(("message" ->> "Node is not ready yet") :: HNil)
-
-  private def validStateForSnapshotReturn(state: NodeState): Boolean = state === NodeState.Ready
+  private val printer = CustomPrinter(dropNullValues = false, indent = "")
+  private val jsonContentType: `Content-Type` =
+    `Content-Type`(MediaType.application.json, Charset.`UTF-8`)
+
+  /** Safe blocking serialization (runs on blocking pool) */
+  private def serializeJson[A: Encoder](value: A): F[Array[Byte]] =
+    Async[F].blocking(printer.printToByteArray(value.asJson, StandardCharsets.UTF_8))
```

Comment:
Did you compare this directly to sync.delay ? Afaik this is CPU only, so its not 'supposed' to be async, but we may have another bug that makes this better to do async